README.md

```diff
@@ -799,16 +530,178 @@ a:hover {
+
+Order classnames in a class attribute by order of "inheritance" -- order in the class attribute does not matter functionally, but helps make the inheritance chain more apparent. 
+
+Separate multiple classes with TWO spaces for readability.
+
+```html
+<!-- Good -->
+<div class="c-tutorial-nav  c-tutorial-nav--alt  u-pull-left">
+
+<!-- Bad - don't use single space -->
+<div class="c-tutorial-nav c-tutorial-nav--alt u-pull-left">
+
+<!-- Bad - not ordered by inheritance: "alt" extends the base. -->
+<div class="c-tutorial-nav--alt  c-tutorial-nav  u-pull-left">
+```
+
+The classes should follow this order:
+
+```html
+<div class="[block|element]  [modifier  [modifier]]  [state]  [utility]  [javascript hooks]  [test hooks]">
+```
+
+
+
+
+
+<a name="single-responsibility-principle"></a>
+### Single Responsibility Principle
+
+Classes should focus on a single thing in order to be reusable; classes that are too specific can often be broken up into multiple blocks, or a block and one or more modifiers.
+
+
+
+<a name="separate-structure-from-skin"></a>
+### Separate Structure from Skin
+
+Component base classes (blocks) should provide structure (e.g., grid layout). They solve the hard problems that we encounter again and again.
+
+Skins (modifiers) should determine more cosmetic aspects (color, font, border, background). In some cases there may be a need for cosmetic styles in the base class as well.
+
+
+
+<a name="separate-containers-from-content"></a>
+### Separate Containers from Content
+
+A component should have no knowledge of its container.
+
+Avoid location-based styles:
+
+```css
+/* Bad: location-based (also: ID selector!) */
+.c-score {font-size: 1.1rem;}
+.l-footer .score {font-size: 1.4rem;}
+
+/* Good */
+.c-score {font-size: 1.1rem;}
+.c-score--lg {font-size: 1.4rem;}
+```
+
+`.score--lg` is a modifier that is placed on the same element as `.score`.
+
+
+
+<a name="component-width-and-height"></a>
+#### Component Width and Height
+
+* Containers (grids, modules) dictate the width of their contents.
+  * Avoid setting widths on anything else.
+* Content dictates the height of its container.
+  * Avoid setting heights on containers.
+
+Components should be flexible and their widths controlled by grids.
+
+```css
+/* Good - dimensions unspecified */
+.c-callout {
+  display: block;
+  border: 1px solid #ccc;
+}
+
+/* Bad - dimensions specified */
+.c-callout {
+  display: block;
+  width: 200px;
+  height: 150px;
+  border: 1px solid #ccc;
+}
+```
+
+A couple limited exceptions to this guideline are constraining the size of an image or icon to fix a specific use case or to limit the length of lines of text to improve readability. If possible, only limit one dimension. Proceed carefully when defining dimensions as it limits the flexibility of the UI to adapt to various device screen sizes.
```

Comment:
What about a note about setting `max` and `min` dimensions instead of just using `width` and `height`? That would improve flexibility while still enforcing limitations only when needed. That may seem obvious to us, but to someone new to CSS it's less clear that it is a reasonable solution.